core/src/services/routing.js

```diff
@@ -659,10 +660,17 @@ class RoutingClass {
           url.hash = url.hash.slice(0, queryParamIndex);
         }
         url.hash = `${url.hash}${queryParamSeparator}${RoutingHelpers.encodeParams(params)}`;
+        if (!url.hash.includes('historyState')) {
+          url.hash += `&historyState=${history.length}`;
+        }
       } else {
-        url.search = `?${RoutingHelpers.encodeParams(params)}`;
+        const search = new URLSearchParams(url.search);
+        if (!search.get('historyState')) {
```

Comment:
Could this be an issue if you have 
**http://localhost:4200/projects/pr2/settings?~foo=historyState........&historyState=5**